In [1]:
import pandas as pd
import numpy as np
import matplotlib
import matplotlib.pyplot as plt
import seaborn as sns

In [6]:
films=pd.read_csv('..\data/films_updated.csv')

In [8]:
pd.set_option('display.max_columns', None)
pd.set_option('display.max_row', None)

In [9]:
films.head()

영화명      영화코드         개봉일 대표국적  국적          장르        등급 영화형태     영화구분  \
0      워킹걸  20147947  2015-01-07   한국  한국  코미디,멜로/로맨스   청소년관람불가   장편     일반영화   
1       설해  20124398  2015-01-08   한국  한국  멜로/로맨스,드라마  12세이상관람가   장편  독립/예술영화   
2   오늘의 연애  20142407  2015-01-14   한국  한국  멜로/로맨스,코미디  15세이상관람가   장편     일반영화   
3      허삼관  20131262  2015-01-14   한국  한국         드라마  12세이상관람가   장편     일반영화   
4  고양이 장례식  20144601  2015-01-15   한국  한국      멜로/로맨스  12세이상관람가   장편  독립/예술영화   

                 제작사                        배급사   상영횟수  전국스크린수         전국매출액  \
0      (주)홍필름,(주)수필름              메가박스중앙(주)플러스엠   7077     308  1.183887e+09   
1           (주)맑음영화사  (주)마인스 엔터테인먼트,주식회사 마운틴픽쳐스    335      34  2.076750e+07   
2          (주)팝콘에프앤엠                  씨제이이앤엠(주)  47237     709  1.492465e+10   
3  (주)두타연,(주)판타지오픽쳐스        (주)넥스트엔터테인먼트월드(NEW)  27661     621  7.405416e+09   
4   영화 고양이 장례식 제작 조합                   (주)인디플러그    394      46  2.320090e+07   

     전국관객수       서울매출액   서울관객수  F-rated   감독 대표감독      감독코드 감독성별 감독_대표역할  \
0   149181   389163479   49874        1  정범식  정범식  10061758   남자      감독   
1     2796    10191000    1317        0  김정권  김정권  10005546   남자      감독   
2  1891993  3951958987  480061        0  박진표  박진표  10029363   남자      감독   
3   955175  1879461269  232983        0  하정우  하정우  10087253   남자      배우   
4     2865    17496200    2146        0  이종훈  이종훈  10057421   남자      감독   

                                                감독필모        작가 대표작가 대표작가(필명)  \
0  공포체험단 곤지암편|천문: 하늘에 묻는다|곤지암|워킹걸|미쓰GO|외톨이|기담|무서운...       정범식  정범식      정범식   
1                설해|그 남자의 책 198쪽|바보|화성으로 간 사나이|동감|귀소  김정권, 김군영  김정권      김정권   
2  오늘의 연애|공범|내 사랑 내 곁에|그놈 목소리|너는 내 운명|죽어도 좋아|신비한 ...  박진표, 이병헌  박진표      박진표   
3  백두산|PMC: 더 벙커|클로젯|신과함께-인과 연|1987|신과함께-죄와 벌|서서평...  하정우, 김주호  하정우      하정우   
4                                       고양이 장례식|심판의날       이종훈  이종훈      이종훈   

       작가코드 작가성별 작가_대표역할                                               작가필모  \
0  10061758   남자      감독  공포체험단 곤지암편|천문: 하늘에 묻는다|곤지암|워킹걸|미쓰GO|외톨이|기담|무서운...   
1  10005546   남자      감독                설해|그 남자의 책 198쪽|바보|화성으로 간 사나이|동감|귀소   
2  10029363   남자      감독  오늘의 연애|공범|내 사랑 내 곁에|그놈 목소리|너는 내 운명|죽어도 좋아|신비한 ...   
3  10087253   남자      배우  백두산|PMC: 더 벙커|클로젯|신과함께-인과 연|1987|신과함께-죄와 벌|서서평...   
4  10057421   남자      감독                                       고양이 장례식|심판의날   

                                                  배우 주연배우  배역명      배우코드 배우성별  \
0  조여정,클라라,김태우,배유람,정범식,정범식,박주희,백경인,조은영,손진우,배성우,유찬...  조여정  백보희  10066680   여자   
1  박해진,이영아,송재희,전무송,최원주,이상민,이기열,윤기호,김수환,김정권,김혜진,박재...  박해진   상우  20126329   남자   
2  이승기,문채원,이서진,장진우,이경진,이동진,임하룡,이상협,김광규,김부선,가인,임종윤...  이승기   준수  10056227   남자   
3  하정우,하지원,전혜진,장광,주진모,성동일,이경영,김영애,정만식,조진웅,김기천,김성균...  하정우  허삼관  10087253   남자   
4      강인,박세영,정겨운,강신유,신무호,김민지,이민희,남기준,민지,김병춘,김도영,차민지   강인  NaN  10005001   남자   

  배우_대표역할                                               배우필모  
0      배우            기생충|워킹걸|인간중독|표적|후궁 : 제왕의 첩|방자전|흡혈형사 나도열  
1      배우                                      치즈인더트랩|설해|젓가락  
2      배우                                          궁합|오늘의 연애  
3      배우  백두산|PMC: 더 벙커|클로젯|신과함께-인과 연|1987|신과함께-죄와 벌|서서평...  
4      배우                 고양이 장례식|순정만화|앨빈과 슈퍼밴드|꽃미남 연쇄 테러 사건

# 성별에 따른 감독 제작비 구하기
전국관객수 기준으로 성별에 따른 TOP5 영화의 값을 찾는다. 그 후, 인터넷 검색을 통해 제작비를 비교할 예정.

In [32]:
start=films.sort_values(by=['전국관객수'],axis=0, ascending=False)

In [33]:
start.head()

영화명      영화코드         개봉일 대표국적  국적       장르        등급 영화형태  영화구분  \
592       극한직업  20182530  2019-01-23   한국  한국      코미디  15세이상관람가   장편  일반영화   
425  신과함께-죄와 벌  20150976  2017-12-20   한국  한국  판타지,드라마  12세이상관람가   장편  일반영화   
70         베테랑  20148048  2015-08-05   한국  한국   액션,드라마  15세이상관람가   장편  일반영화   
66          암살  20148851  2015-07-22   한국  한국   액션,드라마  15세이상관람가   장편  일반영화   
514  신과함께-인과 연  20186202  2018-08-01   한국  한국  판타지,드라마  12세이상관람가   장편  일반영화   

                                 제작사                배급사    상영횟수  전국스크린수  \
592  (주)어바웃필름,영화사 해그림 주식회사,씨제이이앤엠(주)          씨제이이앤엠(주)  292578    1978   
425           리얼라이즈픽쳐스(주),(주)덱스터스튜디오      롯데쇼핑㈜롯데엔터테인먼트  213315    1912   
70                   (주)외유내강,(주)필름케이          씨제이이앤엠(주)  199307    1064   
66                          (주)케이퍼필름             (주)쇼박스  175213    1519   
514           리얼라이즈픽쳐스(주),(주)덱스터스튜디오  롯데컬처웍스(주)롯데엔터테인먼트  180042    2235   

            전국매출액     전국관객수        서울매출액    서울관객수  F-rated   감독 대표감독  \
592  1.396480e+11  16264944  31858660536  3638287        0  이병헌  이병헌   
425  1.156990e+11  14410754  27530825087  3346172        0  김용화  김용화   
70   1.050250e+11  13395400  26766369950  3290366        0  류승완  류승완   
66   9.846313e+10  12705700  25306651181  3155960        1  최동훈  최동훈   
514  1.024770e+11  12253247  22905551759  2636542        0  김용화  김용화   

         감독코드 감독성별 감독_대표역할                                               감독필모  \
592  20125862   남자      감독  드림(가제)|극한직업|레슬러|바람 바람 바람|스물|오늘의 연애|타짜-신의 손|힘내세...   
425  10005151   남자      감독  백두산|신과함께-인과 연|신과함께-죄와 벌|미스터 고|마이 웨이|국가대표 완결판-못...   
70   10019069   남자      감독  엑시트|사바하|군함도|군함도 감독판|베테랑|베를린|평양성|부당거래|해결사|다찌마와리...   
66   10072187   남자      감독  도청(가제)|암살|도둑들|전우치|중천|타짜|소년, 천국에 가다|그때 그 사람들|범죄...   
514  10005151   남자      감독  백두산|신과함께-인과 연|신과함께-죄와 벌|미스터 고|마이 웨이|국가대표 완결판-못...   

           작가 대표작가 대표작가(필명)      작가코드 작가성별   작가_대표역할  \
592       문충일  문충일      문충일  20303529   남자  시나리오(각본)   
425       김용화  김용화      김용화  10005151   남자        감독   
70        류승완  류승완      류승완  10019069   남자        감독   
66   이기철, 최동훈  이기철      이기철  20128927   남자  시나리오(각본)   
514  김용화, 강지원  김용화      김용화  10005151   남자        감독   

                                                  작가필모  \
592                                               극한직업   
425  백두산|신과함께-인과 연|신과함께-죄와 벌|미스터 고|마이 웨이|국가대표 완결판-못...   
70   엑시트|사바하|군함도|군함도 감독판|베테랑|베를린|평양성|부당거래|해결사|다찌마와리...   
66                              도청(가제)|암살|도둑들|가위|난시청지역   
514  백두산|신과함께-인과 연|신과함께-죄와 벌|미스터 고|마이 웨이|국가대표 완결판-못...   

                                                    배우 주연배우  배역명      배우코드  \
592      류승룡,이하늬,진선규,이동휘,공명,양현민,장진희,허준석,한준우,김명준,나철,정재광  류승룡  고반장  10019065   
425  하정우,차태현,주지훈,김향기,마동석,김동욱,도경수,오달수,임원희,장광,정해균,김수안...  하정우   강림  10087253   
70   황정민,유아인,유해진,오달수,장윤주,김시후,오대환,정웅인,정만식,송영창,진경,유인영...  황정민  서도철  10090290   
66   전지현,이정재,하정우,오달수,조진웅,이경영,최덕문,김의성,박병은,김홍파,강용빈,홍성...  전지현  안옥윤  10061467   
514  하정우,주지훈,김향기,마동석,김동욱,이정재,김기태,오희준,도경수,이준혁,이수봉,이용...  하정우  도금연  10087253   

    배우성별 배우_대표역할                                               배우필모  
592   남자      배우  입술은 안돼요(가제)|인생은 아름다워(가제)|극한직업|7년의 밤|염력|제5열|서울역...  
425   남자      배우  백두산|PMC: 더 벙커|클로젯|신과함께-인과 연|1987|신과함께-죄와 벌|서서평...  
70    남자      배우  인질(가제)|다만 악에서 구하소서|공작|군함도|군함도 감독판|아수라|곡성|검사외전|...  
66    여자      배우  암살|베를린|도둑들|도둑들 영화를 만들다|블러드|슈퍼맨이었던 사나이|데이지|내 여자...  
514   남자      배우  백두산|PMC: 더 벙커|클로젯|신과함께-인과 연|1987|신과함께-죄와 벌|서서평...

In [57]:
start_man= start[start['감독성별']=='남자']

In [58]:
start_man.head(5)

영화명      영화코드         개봉일 대표국적  국적       장르        등급 영화형태  영화구분  \
592       극한직업  20182530  2019-01-23   한국  한국      코미디  15세이상관람가   장편  일반영화   
425  신과함께-죄와 벌  20150976  2017-12-20   한국  한국  판타지,드라마  12세이상관람가   장편  일반영화   
70         베테랑  20148048  2015-08-05   한국  한국   액션,드라마  15세이상관람가   장편  일반영화   
66          암살  20148851  2015-07-22   한국  한국   액션,드라마  15세이상관람가   장편  일반영화   
514  신과함께-인과 연  20186202  2018-08-01   한국  한국  판타지,드라마  12세이상관람가   장편  일반영화   

                                 제작사                배급사    상영횟수  전국스크린수  \
592  (주)어바웃필름,영화사 해그림 주식회사,씨제이이앤엠(주)          씨제이이앤엠(주)  292578    1978   
425           리얼라이즈픽쳐스(주),(주)덱스터스튜디오      롯데쇼핑㈜롯데엔터테인먼트  213315    1912   
70                   (주)외유내강,(주)필름케이          씨제이이앤엠(주)  199307    1064   
66                          (주)케이퍼필름             (주)쇼박스  175213    1519   
514           리얼라이즈픽쳐스(주),(주)덱스터스튜디오  롯데컬처웍스(주)롯데엔터테인먼트  180042    2235   

            전국매출액     전국관객수        서울매출액    서울관객수  F-rated   감독 대표감독  \
592  1.396480e+11  16264944  31858660536  3638287        0  이병헌  이병헌   
425  1.156990e+11  14410754  27530825087  3346172        0  김용화  김용화   
70   1.050250e+11  13395400  26766369950  3290366        0  류승완  류승완   
66   9.846313e+10  12705700  25306651181  3155960        1  최동훈  최동훈   
514  1.024770e+11  12253247  22905551759  2636542        0  김용화  김용화   

         감독코드 감독성별 감독_대표역할                                               감독필모  \
592  20125862   남자      감독  드림(가제)|극한직업|레슬러|바람 바람 바람|스물|오늘의 연애|타짜-신의 손|힘내세...   
425  10005151   남자      감독  백두산|신과함께-인과 연|신과함께-죄와 벌|미스터 고|마이 웨이|국가대표 완결판-못...   
70   10019069   남자      감독  엑시트|사바하|군함도|군함도 감독판|베테랑|베를린|평양성|부당거래|해결사|다찌마와리...   
66   10072187   남자      감독  도청(가제)|암살|도둑들|전우치|중천|타짜|소년, 천국에 가다|그때 그 사람들|범죄...   
514  10005151   남자      감독  백두산|신과함께-인과 연|신과함께-죄와 벌|미스터 고|마이 웨이|국가대표 완결판-못...   

           작가 대표작가 대표작가(필명)      작가코드 작가성별   작가_대표역할  \
592       문충일  문충일      문충일  20303529   남자  시나리오(각본)   
425       김용화  김용화      김용화  10005151   남자        감독   
70        류승완  류승완      류승완  10019069   남자        감독   
66   이기철, 최동훈  이기철      이기철  20128927   남자  시나리오(각본)   
514  김용화, 강지원  김용화      김용화  10005151   남자        감독   

                                                  작가필모  \
592                                               극한직업   
425  백두산|신과함께-인과 연|신과함께-죄와 벌|미스터 고|마이 웨이|국가대표 완결판-못...   
70   엑시트|사바하|군함도|군함도 감독판|베테랑|베를린|평양성|부당거래|해결사|다찌마와리...   
66                              도청(가제)|암살|도둑들|가위|난시청지역   
514  백두산|신과함께-인과 연|신과함께-죄와 벌|미스터 고|마이 웨이|국가대표 완결판-못...   

                                                    배우 주연배우  배역명      배우코드  \
592      류승룡,이하늬,진선규,이동휘,공명,양현민,장진희,허준석,한준우,김명준,나철,정재광  류승룡  고반장  10019065   
425  하정우,차태현,주지훈,김향기,마동석,김동욱,도경수,오달수,임원희,장광,정해균,김수안...  하정우   강림  10087253   
70   황정민,유아인,유해진,오달수,장윤주,김시후,오대환,정웅인,정만식,송영창,진경,유인영...  황정민  서도철  10090290   
66   전지현,이정재,하정우,오달수,조진웅,이경영,최덕문,김의성,박병은,김홍파,강용빈,홍성...  전지현  안옥윤  10061467   
514  하정우,주지훈,김향기,마동석,김동욱,이정재,김기태,오희준,도경수,이준혁,이수봉,이용...  하정우  도금연  10087253   

    배우성별 배우_대표역할                                               배우필모  
592   남자      배우  입술은 안돼요(가제)|인생은 아름다워(가제)|극한직업|7년의 밤|염력|제5열|서울역...  
425   남자      배우  백두산|PMC: 더 벙커|클로젯|신과함께-인과 연|1987|신과함께-죄와 벌|서서평...  
70    남자      배우  인질(가제)|다만 악에서 구하소서|공작|군함도|군함도 감독판|아수라|곡성|검사외전|...  
66    여자      배우  암살|베를린|도둑들|도둑들 영화를 만들다|블러드|슈퍼맨이었던 사나이|데이지|내 여자...  
514   남자      배우  백두산|PMC: 더 벙커|클로젯|신과함께-인과 연|1987|신과함께-죄와 벌|서서평...

# 남자 감독 top5 참고: 제목_감독이름 (관객수)
- 극한직업_이병헌(16264944)
- 신과 함께 죄와 벌_김용화 (14410754)
- 베테랑_류승완 (13395400)
- 암살_최동윤 (12705700)
- 신과 함께 인과 연_김용화 (12253247)

In [59]:
start_woman= start[start['감독성별']=='여자']

In [60]:
start_woman.head(5)

영화명      영화코드         개봉일 대표국적  국적      장르        등급 영화형태  영화구분  \
615        돈  20176103  2019-03-20   한국  한국      범죄  15세이상관람가   장편  일반영화   
496  탐정: 리턴즈  20178161  2018-06-13   한국  한국  코미디,범죄  15세이상관람가   장편  일반영화   
586      말모이  20184105  2019-01-09   한국  한국     드라마  12세이상관람가   장편  일반영화   
452  리틀 포레스트  20170841  2018-02-28   한국  한국     드라마     전체관람가   장편  일반영화   
400      부라더  20172603  2017-11-02   한국  한국     코미디  12세이상관람가   장편  일반영화   

                     제작사                배급사    상영횟수  전국스크린수         전국매출액  \
615  (주)사나이픽처스,(주)영화사 월광             (주)쇼박스  120808    1431  2.886961e+10   
496             (주)크리픽쳐스          씨제이이앤엠(주)  107909    1041  2.686572e+10   
586               더램프(주)  롯데컬처웍스(주)롯데엔터테인먼트   87841    1203  2.368264e+10   
452             (주)영화사수박      메가박스중앙(주)플러스엠   68422     884  1.192482e+10   
400        (주)홍필름,(주)수필름      메가박스중앙(주)플러스엠   54976     889  1.159091e+10   

       전국관객수       서울매출액   서울관객수  F-rated   감독 대표감독      감독코드 감독성별   감독_대표역할  \
615  3389035  7287586982  833513        2  박누리  박누리  20151353   여자       조감독   
496  3148038  5989673378  680077        1  이언희  이언희  10056602   여자        감독   
586  2812444  5900912990  681266        2  엄유나  엄유나  20187477   여자  시나리오(각본)   
452  1505560  3853015885  474626        2  임순례  임순례  10058650   여자        감독   
400  1468301  2357877298  287039        2  장유정  장유정  20110174   여자        감독   

                                                  감독필모        작가 대표작가  \
615  돈|남자가 사랑할 때|베를린|부당거래|서울이 보이냐|더 게임|김관장 대 김관장 대 김관장       박누리  박누리   
496  탐정: 리턴즈|미씽: 사라진 여자|타이밍|어깨 너머의 연인|해피 에로 크리스마스|....       정한진  정한진   
586                                   말모이|택시운전사|번개와 춤을       엄유나  엄유나   
452  리틀 포레스트|글로리데이|제보자|남쪽으로 튀어|로맨스 조|소와 함께 여행하는 법|날...       황성구  황성구   
400                                  정직한 후보|부라더|김종욱 찾기  장유정, 허성혜  장유정   

    대표작가(필명)      작가코드 작가성별   작가_대표역할  \
615      박누리  20151353   여자       조감독   
496      정한진  20279790   남자  시나리오(각본)   
586      엄유나  20187477   여자  시나리오(각본)   
452      황성구  10090124   남자  시나리오(각본)   
400      장유정  20110174   여자        감독   

                                                  작가필모  \
615  돈|남자가 사랑할 때|베를린|부당거래|서울이 보이냐|더 게임|김관장 대 김관장 대 김관장   
496                                            탐정: 리턴즈   
586                                   말모이|택시운전사|번개와 춤을   
452  리틀 포레스트|박열|찌라시: 위험한 소문|나는 왕이로소이다|간기남|특수본|식객2 :...   
400                                  정직한 후보|부라더|김종욱 찾기   

                                                    배우 주연배우  배역명      배우코드  \
615                류준열,유지태,조우진,원진아,박명신,이화룡,오희준,정기섭,차래형  류준열  조일현  20208626   
496  권상우,성동일,이광수,서영희,이일화,남명렬,손담비,최성원,김동욱,박성일,오희준,이광...  권상우   도희  10002387   
586  유해진,윤계상,김홍파,우현,김태훈,김선영,민진웅,이성욱,조현철,오희준,정기섭,허성태...  유해진  박종환  10054128   
452  김태리,류준열,문소리,진기주,홍성덕,이광용,이상준,김재훈,이유진,김정은,김주은,이태...  김태리   유정  20201026   
400  마동석,이동휘,이하늬,류정수,배준수,석정호,신은영,안백승,안승주,채지혜,이성은,이수...  마동석   현정  10021341   

    배우성별 배우_대표역할                                               배우필모  
615   남자      배우  봉오동 전투|돈|뺑반|리틀 포레스트|독전: 익스텐디드 컷|독전|침묵|택시운전사|더 ...  
496   남자      배우  신의 한 수: 귀수편|두번할까요|히트맨|탐정: 리턴즈|탐정 : 더 비기닝|그림자 애...  
586   남자      배우  봉오동 전투|승리호(가제)|말모이|완벽한 타인|레슬러|1987|택시운전사|공조|럭키...  
452   여자      배우  승리호(가제)|리틀 포레스트|1987|아가씨|아가씨(확장판)|문영|문영 단편|락 아...  
400   남자      배우  나쁜 녀석들: 더 무비|노리개: 그녀의 눈물|악인전|백두산|시동|성난황소|동네사람들...

# 여자 감독 top5 참고: 제목_감독이름 (관객수)
- 돈_박누리(3389035)
- 탐정:리턴즈_이언희 (3148038)
- 말모이_엄유나 (2812444)
- 리틀포레스트_엄순례 (1505560)
- 부라더_장유정 (1468301)

# 정리

# 남자 감독 top5 참고: 제목_감독이름 (관객수)
- 극한직업_이병헌(16264944)
- 신과 함께 죄와 벌_김용화 (14410754)
- 베테랑_류승완 (13395400)
- 암살_최동윤 (12705700)
- 신과 함께 인과 연_김용화 (12253247)

# 여자 감독 top5 참고: 제목_감독이름 (관객수)
- 돈_박누리(3389035)
- 탐정:리턴즈_이언희 (3148038)
- 말모이_엄유나 (2812444)
- 리틀포레스트_엄순례 (1505560)
- 부라더_장유정 (1468301)

In [61]:
money=pd.read_csv('..\data/movie_money_power.csv')

성별에 따른 흥행작품의 손익분기점과 제작비 입력. 
* 제작비는 '억' 단위. 
* 관객수- 손익분기점 이름을 무엇으로 정할 지 몰라서 일단 이름...은 이렇게 설정함.

In [63]:
money.head()

영화명  제작비       관객수    손익분기점  관객수 - 손익분기 대표감독 감독성별
0       극한직업   60  16264944  2000000    14264944  이병헌   남자
1  신과함께-죄와 벌  400  14410754  6000000     8410754  김용화   남자
2        베테랑   60  13395400  2800000    10595400  류승완   남자
3         암살  220  12705700  6500000     6205700  최동윤   남자
4  신과함께-인과 연  400  12253247  6000000     6253247  김용화   남자

In [65]:
money

영화명  제작비       관객수    손익분기점  관객수 - 손익분기 대표감독 감독성별
0       극한직업   60  16264944  2000000    14264944  이병헌   남자
1  신과함께-죄와 벌  400  14410754  6000000     8410754  김용화   남자
2        베테랑   60  13395400  2800000    10595400  류승완   남자
3         암살  220  12705700  6500000     6205700  최동윤   남자
4  신과함께-인과 연  400  12253247  6000000     6253247  김용화   남자
5          돈   80   3389035  2000000     1389035  박누리   여자
6     탐정:리턴즈   50   3148038  1800000     1348038  이언희   여자
7        말모이  110    281244  3000000    -2718756  엄유나   여자
8     리틀포레스트   15   1505560   800000      705560  엄순례   여자
9        부라더   20   1468301  1000000      468301  장유정   여자

In [ ]:
# 엑셀로 대충 작업 함.